File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.java
class Solution {
    private final int MODULO = 1_000_000_007;
    
    public int countNicePairs(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            nums[i] -= rev(nums[i]);
        }
        HashMap<Integer, Integer> count = new HashMap<>();
        int total = 0;
        for (int i = 0; i < nums.length; i++) {
            total += count.getOrDefault(nums[i], 0);
            total %= MODULO;
            count.merge(nums[i], 1, Integer::sum);
        }
        return total;
        
    }
    
    private int rev(int x) {
        int acc = 0;
        for (int y = x; y > 0; y /= 10) {
            acc *= 10;
            acc += y % 10;
        }
        return acc;
    }
}

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        def rev(x, acc=0): return acc if not x else rev(x // 10, acc * 10 + x % 10)
        counter = Counter(map(lambda x: x - rev(x), nums))
        return sum(c * (c - 1) // 2 for c in counter.values()) % 1_000_000_007
        